### ISSUE TITLE
Checkbox label

### ISSUE THREAD

**Issue by andresz1** [adevinta/spark#534]
### Description

The checkbox component should have a label integrated

### Problem Statement/Justification

The checkbox component is missing a label and it's needed in terms of accessibility. ATM could be added by the user but to provide a better DX I think it should be provided out of the box. 

### Proposed Solution or API

```tsx
<Checkbox> {/* label */}
  <CheckboxInput /> {/* input */}

  <CheckboxText> {/* span */}
     Groceries
  </CheckboxText>
</Checkbox>
```
